src/modules/esl-utils/dom/events/listener.ts

```diff
@@ -0,0 +1,148 @@
+import {sequentialUID} from '../../misc/uid';
+import {memoize} from '../../decorators/memoize';
+import {isSimilar} from '../../misc/object/compare';
+import {TraversingQuery} from '../../../esl-traversing-query/core';
+
+/** Describes callback handler */
+export type ESLListenerHandler<EType extends Event = Event> = (event: EType, listener?: ESLEventListener) => void;
+
+/** Descriptor to create {@link ESLEventListener} */
+export type ESLListenerDescriptor<EType extends string = string> = {
+  /** Event type (name) */
+  event: EType;
+  /** Use capture DOM Event phase */
+  capture?: boolean;
+
+  /** CSS selector to check delegated event */
+  selector?: string;
+  /** {@link TraversingQuery} selector or element target to subscribe */
+  target?: string | EventTarget;
+
+  /** Identifier to group event listeners */
+  id?: string;
+  /** Marks the listener to be automatically subscribed and unsubscribed within connected/disconnected elements hooks */
+  auto?: boolean;
+};
+
+/** Condition (criteria) to find {@link ESLListenerDescriptor} */
+export type ESLListenerCriteria = undefined | string | ESLListenerHandler | Partial<ESLListenerDescriptor>;
+
+/** Function decorated as {@link ESLListenerDescriptor} */
+export type ESLListenerDescriptorFn = ESLListenerHandler & ESLListenerDescriptor;
+
+/** Type guard to check if the passed function is typeof {@link ESLListenerDescriptorFn} */
+export const isDescriptorFn = (obj: any): obj is ESLListenerDescriptorFn => typeof obj === 'function' && typeof obj.event === 'string';
+
+const STORE = '__listeners';
+
+/** Event Listener instance, used as an 'inner' record to process subscriptions made by `EventUtils` */
+export class ESLEventListener implements ESLListenerDescriptor {
+  public readonly id: string;
+  public readonly event: string;
+  public readonly capture?: boolean;
+  public readonly selector?: string;
+  public readonly target?: string | EventTarget;
+
+  constructor(
+    public readonly $host: HTMLElement,
+    public readonly handler: ESLListenerHandler,
+    desc: ESLListenerDescriptor
+  ) {
+    desc.id = desc.id || sequentialUID('esl.event');
+    Object.assign(this, {capture: false}, desc);
+    this.handle = this.handle.bind(this);
```

Comment:
Yep, of corse.
The `@bind` decorator uses speciall tricks to manage method binding. Such a tricks is +/- lightweight to use it free in the consumers code. 
But here we have a core mechanism part that needs to be as much effective as it's possible. So in case we defining library core I'd prefer to relay on the faster and simpler tools then syntax sugar

The only exclusion I afford here is `@memoize` to cache  targets to be able to unsubscribe them safe. And the only reason for that is that instance getters uses super fast API while manual cashe implementation is a little bit bulky.